components/script/dom/webgl2renderingcontext.rs

```diff
@@ -1721,6 +1745,214 @@ impl WebGL2RenderingContextMethods for WebGL2RenderingContext {
+    }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.15
+    fn DeleteTransformFeedback(&self, tf: Option<&WebGLTransformFeedback>) {
+        if let Some(tf) = tf {
+            handle_potential_webgl_error!(self.base, self.base.validate_ownership(tf), return);
+            if tf.is_active() {
+                self.base.webgl_error(InvalidOperation);
+                return;
+            }
+            tf.delete(false);
+            self.current_transform_feedback.set(None);
+        }
+    }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.15
+    fn IsTransformFeedback(&self, tf: Option<&WebGLTransformFeedback>) -> bool {
+        match tf {
+            Some(tf) => {
+                if !tf.is_valid() {
+                    return false;
+                }
+                handle_potential_webgl_error!(
+                    self.base,
+                    self.base.validate_ownership(tf),
+                    return false
+                );
+                let (sender, receiver) = webgl_channel().unwrap();
+                self.base
+                    .send_command(WebGLCommand::IsTransformFeedback(tf.id(), sender));
+                receiver.recv().unwrap()
+            },
+            None => false,
+        }
+    }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.15
+    fn BindTransformFeedback(&self, target: u32, tf: Option<&WebGLTransformFeedback>) {
+        if target != constants::TRANSFORM_FEEDBACK {
+            self.base.webgl_error(InvalidEnum);
+            return;
+        }
+        match tf {
+            Some(transform_feedback) => {
+                handle_potential_webgl_error!(
+                    self.base,
+                    self.base.validate_ownership(transform_feedback),
+                    return
+                );
+                if !transform_feedback.is_valid() {
+                    self.base.webgl_error(InvalidOperation);
+                    return;
+                }
+                if let Some(current_tf) = self.current_transform_feedback.get() {
+                    if current_tf.is_active() && !current_tf.is_paused() {
+                        self.base.webgl_error(InvalidOperation);
+                        return;
+                    }
+                }
+                transform_feedback.bind(&self.base, target);
+                self.current_transform_feedback
+                    .set(Some(transform_feedback));
+            },
+            None => self
+                .base
+                .send_command(WebGLCommand::BindTransformFeedback(target, 0)),
+        }
+    }
+
+    /// https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7.15
+    fn BeginTransformFeedback(&self, primitiveMode: u32) {
+        match primitiveMode {
+            constants::POINTS | constants::LINES | constants::TRIANGLES => {
+                match self.current_transform_feedback.get() {
```

Comment:
nit: we can reduce some rightward drift with:
```rust
match primitiveMode {
    constants::POINTS | constants::LINES | constants::TRIANGLES => {}
    _ => {
        self.base.webgl_error(InvalidEnum);
        return;
    }
}
let current_tf = match self.current_transform_feedback.get() {
    Some(current_tf) => current_tf,
    None => {
        self.base.webgl_error(InvalidEnum);
        return;
    }
};
if current_tf.is_active() {
    self.base.webgl_error(InvalidEnum);
    return;
}
let program = match self.base.current_program() {
    Some(program) => program,
    None => {
        self.base.webgl_error(InvalidOperation);
        return;
    }
};
// etc.
```

Reply:
Thank you! It is more readable this way. I've updated the PR with this change.